### PR TITLE
Feature/structures mode-Updated

### PR DESCRIPTION
**Pull request by theanuradha**

### CODE REVIEW

File: qupath-gui-fx/src/main/java/qupath/lib/gui/viewer/tools/PolygonEditTool.java [hunk 1]
```diff
@@ -146,7 +146,7 @@ public void mouseMoved(MouseEvent e) {
         }
 
 
-        boolean isLocked = ((PathAnnotationObject) pathObject).isLocked();
+        boolean isLocked = ((PathAnnotationObject) pathObject).isLocked() || !checkIfActionValid(viewer.getSelectedObject());
```
Comment (godardt) [inline qupath-gui-fx/src/main/java/qupath/lib/gui/viewer/tools/PolygonEditTool.java:149]: How is this `checkIfActionValid` supposed to help for the structures mode?

File: qupath-gui-fx/src/main/java/qupath/lib/gui/QuPathGUI.java [hunk 2]
```diff
@@ -299,6 +299,7 @@ public void setUserProfileChoice(UserProfileChoice profileChoice) {
 
     public enum UserProfileChoice {
         SPECIALIST_MODE("Specialist mode", 3),
+        STRUCTURES_MODE("Structures mode", 4),
```
Comment (godardt) [inline qupath-gui-fx/src/main/java/qupath/lib/gui/QuPathGUI.java]: The Structures mode stands between the Specialist and contractor mode so its ID should be `3` and the Specialist mode should now be `4`

**Review by godardt — CHANGES_REQUESTED** [review]
Please implement the STRUCTURE_MODE fully before submitting a PR. If I merge now the feature will only work partially without the validation process.

File: qupath-gui-fx/src/main/java/qupath/lib/gui/viewer/tools/PolygonEditTool.java (hunk 1 above)
Comment (theanuradha) [reply qupath-gui-fx/src/main/java/qupath/lib/gui/viewer/tools/PolygonEditTool.java:149]: this is for stop PolygonEditTool from edit ROI or Inflammation if mode is Contract

**Review by godardt — CHANGES_REQUESTED** [review]
Looks great! Only one thing to add: The "Create project" and "Import Mirax files from root path" buttons in Structure mode should be disabled (same as with the Contractor mode)

**Review by godardt — APPROVED** [review]
Perfect thanks :)

**Pull request merged**